src/OrchardCore.Modules/OrchardCore.Media/Drivers/MediaFieldDriver.cs

```diff
@@ -123,6 +145,8 @@ public override async Task<IDisplayResult> UpdateAsync(MediaField field, IUpdate
 
                 field.Paths = items.Where(p => !p.IsRemoved).Select(p => p.Path).ToArray() ?? new string[] { };
 
+                field.SetAttachedFileNames(items.Where(i => !i.IsRemoved).Select(i => i.AttachedFileName).ToArray());
```

Comment:
I think this should only happen when the editor is attached. Around line 115

Reply:
I moved that line up in  https://github.com/OrchardCMS/OrchardCore/pull/10570/commits/5280f52ecdeb677b0b64216b5e99ffa3a1bd8f95